### DISCUSSION TITLE
Support for Gateway API

### DISCUSSION THREAD

**Discussion by shashankram** [projectcontour/contour#3822] [General]
Hi, I am interested in knowing how much of the [Gateway API](https://gateway-api.sigs.k8s.io/) does Contour currently support, and what's the roadmap to provide HTTP/TCP/TLS based ingress using the Gateway API.

My use case is to leverage the Gateway API for the different protocols noted above and proxying to the backend over mTLS.

**Comment by shashankram** [comment]
I found this https://github.com/projectcontour/contour/issues/2287

**Reply by stevesloka** [reply]
Hey @shashankram! That's a great issue to follow as this is very much a moving target. There's also an in-progress update to the design doc which outlines what Contour will support in GatewayAPI (As it stands right now) here: https://github.com/projectcontour/contour/pull/3797

**Reply by shashankram** [reply]
Thanks @stevesloka for sharing the design doc. I'll go through it!
